src/api.rs

```diff
@@ -111,11 +124,26 @@ pub struct PlayersResponse {
 /// This is used by new host connections to update thier display to match the current state of the
 /// game.
 #[get("/players")]
-pub fn get_players(scoreboard: State<Mutex<Scoreboard>>) -> JSON<PlayersResponse> {
+pub fn get_players(
+    scoreboard: State<Mutex<Scoreboard>>,
+    usernames: State<Mutex<Usernames>>,
+) -> JSON<PlayersResponse> {
     // Clone the scoreboard so we can release the lock on it quickly.
     let scoreboard = scoreboard.lock().expect("Scoreboard mutex was poisoned").clone();
 
-    let players = scoreboard.iter().map(|(&id, &score)| PlayerData { id, score }).collect();
+    let usernames = usernames.lock().expect("Usernames mutex was poisoned").clone();
+
+    let players = scoreboard.iter()
+        .map(
+            |(&id, &score)|
+            {
+                let username = match usernames.get(&id) {
+                    None => "".to_string(),
+                    Some(i) => i.to_string(),
+                };
```

Comment:
In `register_player` you guarantee that any registered player will have a username. As such, `usernames.get(&id)` returning `None` when `scoreboard.get(&id)` returned `Some` (i.e. the player has a score but no name) indicates that there is a bug in the app. In this case, you silently ignore that bug. It would be better to instead `panic!`, which would help us catch and debug the issue. To do so, write:

```
let username = usernames.get(&id).expect("Player ID was in scoreboard but not usernames table");
```